src/Val.h

```diff
@@ -763,6 +771,33 @@ class TableVal final : public Val, public notifier::detail::Modifiable {
 	 */
 	TableValPtr Intersection(const TableVal& v) const;
 
+	/**
+	 * Returns a new table that is the union of this table and the
+	 * given table.  Union is done only on index, so this generally
+	 * makes most sense to use for sets, not tables.
+	 * @param v  The union'ing table.
+	 * @return  The union of this table and the given one.
+	 */
+	TableValPtr Union(TableVal* v) const
+		{
+		auto v_clone = cast_intrusive<TableVal>(v->Clone());
+		AddTo(v_clone.get(), false, false);
+		return v_clone;
+		}
+
+	/**
+	 * Returns a copy of this table with the given table removed.
+	 * @param v  The table to remove.
+	 * @return  The subset of this table that doesn't include v.
+	 */
+	TableValPtr TakeOut(TableVal* v)
+		{
+		auto clone = cast_intrusive<TableVal>(Clone());
+		cast_intrusive<TableVal>(v->Clone());
```

Comment:
```suggestion
```
I did this removal during merge since it doesn't seem to do anything except cause a copy to made and immediately discarded, please correct if mistaken.

Reply:
D'oh, not sure what sort of cut-and-paste glitch led to that one, but you're also correct in removing it.